### DISCUSSION TITLE
Metadata API doesnt work with components marked with "use client"

### DISCUSSION THREAD

**Discussion by FilipLusnia** [vercel/next.js#46368] [Help]
As in title - Metadata API doesnt work with components marked with "use client". Instead, they take the metadata from my layout.tsx. When I remove the "use client" directive from component, it suddenly accepts the local metadata object.

What should I do to fix this? Use next/head?

EDIT : 
next/head doesn't work either.

**Comment by djordje-st** [comment]
As of version 13.2 metadata works only on server components such as `page.js` or `layout.js` [(docs)](https://beta.nextjs.org/docs/api-reference/metadata).

The `<Head>` component from `next/head` works only in the `/pages` directory.

That being said it's best if you define metadata in `page.js` or `layout.js` and following the docs. There's support both for static and dynamic metadata listed there,

**Reply by quangkhaidam93** [reply]
You mean that metadata need either `layout.ts` or `page.ts` are server components. Is it right?

**Comment by CEN-Smart** [comment]
When using a UI library like material UI, Mantine, Chakra UI which requires you wrap your entire root element with their Provider, it only works when you make them a client component, while the new update for the metadata API only works for client components, please this conflicts should be looked in or better still make the metadata API accepts both client and server components.

**Reply by shuding** [reply]
You can still use them in a server component:

```jsx
// provider.js
'use client'

import { Provider } from 'design-system'

export default Provider
```

In your root layout (a server component):

```jsx
import Provider from './provider'

export default Layout({ children }) {
  return <Provider>
    <html>
      <body>{children}</body>
    </html>
  </Provider>
}

export const metadata = {
  ...
}
```

So the provider will be a client component that wraps everything.

**Reply by pwrnrd** [reply]
There is probably a good reason why the project has moved to the Metadata API but by making this change Next forces everyone to rewrite their components to support Metadata, especially because the Head file will be deprecated. This is not a small version bump. This is a breaking change.

**Reply by knupel** [reply]
> 
> import { Provider } from 'design-system'
> 
I've the same problem. I'm new with `nextjs`.  
Itry your solution but I don't find where is `design-system` I'm use the last nextjs with `tailwind` this plugin is from it ?

**Comment by Willem-Jaap** [comment]


**Reply by pwrnrd** [reply]
I agree. I don't see a reason why it should not be possible for the new metadata API to work with client components. If metadata is not allowed in client components, the compiler could take care of splitting the metadata into its own file. Now devs need to split components. This forces devs to basically recreate the Head.js file for client side components, but without a standard, making projects across developers less consistent. It would be great if the compiler itself could take care of this.

**Comment by hexionaut** [comment]
I have this problem for example with Google Tag Manager where I need to send data to it with this code :

```js
  const router = useRouter();

  const pageview = (url: string) => {
    // @ts-ignore
    window.dataLayer.push({
      event: 'pageview',
      page: url
    });
  };

  useEffect(() => {
    router.events.on('routeChangeComplete', pageview);
    return () => {
      router.events.off('routeChangeComplete', pageview);
    };
  }, [router.events]);
```

Wanted to put it in my `layout.js` file but I can't use Metadata if I do so.

**Comment by rumon5h** [comment]
After `<HTML>` starting tag on Layout I added Head tag to fix this problem!


```
    <html lang="en">
      <head>
        <title>Title</title>
        <meta name='description' content='Description' />
      </head>
      <body className={`${inter.className} max-w-[1580px] mx-auto`}>
        
      </body>
    </html>
```

**Comment by NisargOp** [comment]
For those who are still facing this issue, follow these simple steps:

1. Create a component or page that includes all the client-side code requiring "use client."
2. Next, navigate to your main page where you want to add the metadata.
3. Remove the old code, import the component or page you just created, and ensure you do not use "use client" in this section.
4. Finally, add your metadata.
![new code structure](https://github.com/user-attachments/assets/2d529e62-4fed-47e1-b847-80f2299a224d)

**Comment by impishlucy** [comment]
Are we serious ? Why shouldnt i be able to build a fully static site with different titles in Next.JS this is actually pathetic.
Stop wasting so much time on your SSR garbage and fix the basics.

**Reply by Willem-Jaap** [reply]
just use the `<title>` tag as a workaround
https://react.dev/reference/react-dom/components/title#reference
